src/models/config/monitor_config.rs

```diff
@@ -74,6 +75,24 @@ impl ConfigLoader for Monitor {
 				.to_string();
 
 			let monitor = Self::load_from_path(&path).await?;
+
+			// Check monitor name uniqueness before pushing
+			if pairs
+				.iter()
+				.any(|(_, existing_monitor): &(String, Monitor)| {
+					normalize_string(&existing_monitor.name) == normalize_string(&monitor.name)
+				}) {
+				return Err(ConfigError::validation_error(
+					format!("Duplicate monitor name found: '{}'", monitor.name),
+					None,
+					Some(HashMap::from([
+						("monitor_name".to_string(), monitor.name.clone()),
+						("filename".to_string(), name),
+						("path".to_string(), path.display().to_string()),
+					])),
+				));
+			}
+
```

Comment:
I'm thinking if it makes sense to move this to `validate` instead?
We have some validation logic there already, including, for example: `validate_protocol`.